src/transformers/modeling_utils.py

```diff
@@ -2188,6 +2189,19 @@ def from_pretrained(
         subfolder = kwargs.pop("subfolder", "")
         commit_hash = kwargs.pop("_commit_hash", None)
         variant = kwargs.pop("variant", None)
+        kwargs.pop("peft_adapter_name", None)
+        peft_adapter_model_id = kwargs.pop("_peft_adapter_model_id", None)
+
+        is_adapter_file = False
+        if is_peft_available():
+            is_adapter_file = peft_adapter_model_id is not None
```

Comment:
Maybe raise an error if someone wants to load a model with an adapter file and peft is not installed? I don't know how common that case is, but it seems to me that the user would want to use the adapter file if it's there no?

Reply:
I think this is a great idea, and I am keen to add that condition as well. However, I am worried about something, this means we'll need to look for the `adapter_config.json` file on the local dir or remote directory in any case and I want to avoid that and restrict only to users that have PEFT installed. What do you think? If you think that this is fine, happy to add it.